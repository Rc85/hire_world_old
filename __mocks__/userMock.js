export const User = {
    account_type: "Listing",
    allow_messaging: true,
    avatar_url: "/user_files/1/businessman-avatar-icon-01-.jpg",
    display_fullname: true,
    email_notifications: false,
    hide_email: false,
    user_address: "3475 E 27th Ave",
    user_bio: "Lorem ipsum dolor sit amet consectetur adipisicing elit. Assumenda ullam repellat maiores aperiam debitis iure, repudiandae maxime pariatur nam rerum quasi numquam aliquid illo temporibus amet officiis, veritatis ipsum? Illum. Lorem ipsum dolor sit amet consectetur adipisicing elit. Aspernatur similique quae tenetur repellat earum voluptate, reiciendis doloribus. Pariatur atque porro sunt autem quod harum soluta necessitatibus tempora hic. Molestias, harum! Lorem ipsum dolor sit amet consectetur adipisicing elit. Hic molestiae nihil, optio accusantium repellat ex cupiditate commodi similique facere minus maxime explicabo dolorum neque veniam corrupti ipsa debitis. Pariatur, necessitatibus. Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ducimus voluptatem, a molestias aperiam obcaecati doloribus harum veritatis suscipit asperiores hic exercitationem molestiae aliquid soluta minima! Laborum placeat saepe dolores! Enim.",
    user_business_name: "RC Web Development",
    user_city: "Vancouver",
    user_city_code: "V5R1P8",
    user_country: "Canada",
    user_education: "BCIT",
    user_email: "rogerchin85@gmail.com",
    user_facebook: "https://www.facebook.com",
    user_firstname: "Roger",
    user_github: "https://github.com/Rc85",
    user_instagram: "https://www.instagram.com",
    user_last_login: "2018-12-10T00:32:16.540Z",
    user_lastname: "Chin",
    user_linkedin: "https://www.linkedin.com",
    user_phone: "604-889-8353",
    user_profile_id: 1,
    user_region: "British Columbia",
    user_setting_id: 1,
    user_title: "Web Developer",
    user_twitter: "https://www.twitter.com",
    user_website: "https://www.rogerchin.me",
    user_worldwide: null,
    username: "roger85"
}