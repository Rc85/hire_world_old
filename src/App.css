.panel-container {
    display: flex;
    justify-content: center;
}

.main-container {
    flex: 1;
    position: relative;
    display: flex;
    justify-content: center;
}

.blue-panel {
    background: #2F496E;
}

.blue-panel.deep {
    padding: 100px;
}

.blue-panel.shallow {
    padding: 25px;
}

.blue-panel.rounded-corners {
    border-radius: 5px;
}

.form-buttons {
    text-align: right;
}

.form-buttons input[type=submit] {
    margin-right: 5px;
}

.section-container {
    margin-bottom: 20px;
}

.main-panel {
    padding: 100px 200px;
    display: flex;
    flex-direction: column;
}

.three-rounded {
    border-bottom-left-radius: 5px;
    border-bottom-right-radius: 5px;
    border-top-right-radius: 5px;
}

.section-container .section-header {
    border-top-left-radius: 5px;
    border-top-right-radius: 5px;
    background: linear-gradient(#406497, #2F496E);
    padding: 15px;
}

.section-container .section-body {
    border-width: 0 1px 1px 1px;
    border-color: #2F496E;
    border-style: solid;
    border-bottom-left-radius: 5px;
    border-bottom-right-radius: 5px;
    padding: 15px;
    color: #000;
}

.list-horizontal {
    display: flex;
    flex-wrap: wrap;
}

.col-container {
    display: flex;
    flex-direction: column;
    width: 100%;
    min-height: 100%;
    position: relative;
}

.row-container {
    display: flex;
    justify-content: space-between;
    width: 100%;
    min-height: 100%;
    flex-wrap: wrap;
}

.section-container {
    width: 17%;
}

.submit-btn {
    width: 90px;
}

.menu-container {
    position: relative;
}

.menu {
    position: absolute;
    background: #C4C4C4;
    z-index: 1;
}

.menu-button {
    cursor: pointer;
}

.menu-item {
    padding: 10px;
    min-width: 75px;
    text-align: center;
    color: #000;
}

.menu-item:hover {
    background: #2988BC;
    cursor: pointer;
    color: #FFF;
}

.menu-item:last-child {
    border: 1px solid #2988BC;
}

.menu-item:not(:last-child) {
    border-width: 1px 1px 0 1px;
    border-color: #2988BC;
    border-style: solid;
}

.tab-bar {
    display: flex;
}

.tab-button {
    border-top-right-radius: 5px;
    border-top-left-radius: 5px;
    border-width: 1px 1px 0 1px;
    border-color: #2F496E;
    border-style: solid;
    padding: 10px;
    color: #000;
    margin-right: 5px;
}

.tab-button:hover {
    background: #2988BC;
    cursor: pointer;
    color: #F9F6F4;
}

.tab-button.active {
    background: #2F496E;
    color: #F9F6F4;
}

.delete-button {
    cursor: pointer;
}

.required-asterisk {
    color: #ED463D;
}

.grey-panel {
    background: #F9F6F4;
    padding: 10px;
    color: #000;
}