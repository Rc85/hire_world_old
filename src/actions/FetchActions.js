import fetch from 'axios';

export const GetSession = () => {
    return dispatch => {
        fetch.post('/api/auth/get-session')
        .then(resp => {
            if (resp.data.status === 'get session success') {
                dispatch(GetSessionSuccess(resp.data.status, resp.data.user));
            }
        })
        .catch(err => console.log(err));
    }
}

const GetSessionSuccess = (status, user) => {
    return {
        type: 'GET_SESSION_SUCCESS',
        user,
        status
    }
}

export const GetSectors = () => {
    return dispatch => {
        fetch.get('/api/get/sectors')
        .then(resp => {
            if (resp.data.status === 'get sectors success') {
                dispatch(StoreSectors(resp.data.status, resp.data.sectors));
            } else {
                dispatch(StoreSectorsError(resp.data.status));
            }
        })
        .catch(err => console.log(err));
    }
}

const StoreSectors = (status, sectors) => {
    return {
        type: 'STORE_SECTOR',
        sectors,
        status
    }
}

const StoreSectorsError = (status) => {
    return {
        type: 'STORE_SECTOR_ERROR',
        status
    }
}

export const GetServices = () => {
    return dispatch => {
        fetch.get('/api/get/services')
        .then(resp => {
            if (resp.data.status === 'get services success') {
                dispatch(UpdateServices(resp.data.status, resp.data.services));
            } else {
                dispatch(UpdateServicesError(resp.data.status));
            }
        });
    }
}

const UpdateServices = (status, services) => {
    return {
        type: 'UPDATE_SERVICES',
        status,
        services
    }
}

const UpdateServicesError = status => {
    return {
        type: 'UPDATE_SERVICES_ERROR',
        status
    }
}