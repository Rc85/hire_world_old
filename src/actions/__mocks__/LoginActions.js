export const User = {
    avatar_url: "../images/profile.png",
    business_name: "",
    default_location: false,
    display_business_name: false,
    display_contacts: false,
    display_fullname: false,
    email_notifications: true,
    hide_email: false,
    services_allowed: 3,
    user_address: null,
    user_bio: null,
    user_city: "",
    user_country: "",
    user_created_on: "2018-08-30T00:20:31.456Z",
    user_education: null,
    user_email: "hacker@hacker.com",
    user_facebook: null,
    user_firstname: "",
    user_github: null,
    user_id: 19,
    user_instagram: null,
    user_lastname: "",
    user_level: 0,
    user_linkedin: null,
    user_phone: null,
    user_region: "",
    user_status: "Active",
    user_title: null,
    user_twitter: null,
    user_type: null,
    user_website: null,
    user_worldwide: false,
    username: "hacker",
    view_counter: 0
}