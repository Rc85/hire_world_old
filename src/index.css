@font-face {
    font-family: 'Montserrat';
    src: url('./fonts/Montserrat-Regular.ttf');
}

@font-face {
    font-family: 'Montserrat Bold';
    src: url('./fonts/Montserrat-Bold.ttf');
}

html {
    min-height: 100vh;
}

body {
    margin: 0;
    padding: 0;
    font-family: 'Montserrat';
    background: #C4C4C4;
    color: #F9F6F4;
    min-height: 100vh;
    display: flex;
    overflow-x: hidden;
}

#root {
    width: 100%;
    min-height: 100%;
}

a {
    font-weight: bold;
    color: #F9F6F4;
}

a:hover {
    color: #F9F6F4;
    text-decoration: underline;
}

h1, h2, h3, h4, h5, h6 {
    font-family: 'Montserrat Bold';
}

hr {
    background: #2988BC;
}

/* Widths */

.w-5 {
    width: 5%;
}

.w-10 {
    width: 10%;
}

.w-20 {
    width: 20%;
}

.w-30 {
    width: 30%;
}

.w-33 {
    width: 33%;
}

.w-35 {
    width: 35%;
}

.w-40 {
    width: 40%;
}

.w-45 {
    width: 45%;
}

.w-55 {
    width: 55%;
}

.w-60 {
    width: 60%;
}
