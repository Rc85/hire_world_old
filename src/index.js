import 'babel-polyfill';
import 'normalize.css';
import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter as Router } from 'react-router-dom';
import './styles/vendor/bootstrap.css';
import './styles/vendor/bootstrap-grid.css';
import './styles/vendor/bootstrap-reboot.css';
import './styles/index.css';
import App from './App';
import { Provider } from 'react-redux';
import { reducers } from './reducers';
import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';

const store = createStore(reducers, applyMiddleware(thunk));
store.subscribe(() => { console.log(store.getState()) });

ReactDOM.render(
    <Provider store={store}>
        <Router basename='/'>
            <App />
        </Router>
    </Provider>,
    document.getElementById('root')
);