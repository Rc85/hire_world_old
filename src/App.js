import React, { Component } from 'react';
import { Route, withRouter, Switch } from 'react-router-dom';
import { connect } from 'react-redux';
import { GetSession, GetSectors } from './actions/FetchActions';
import { RemoveAlert } from './actions/AlertActions';
import * as Pages from './components/pages';
import * as Admin from './components/admin';
import TopBar from './components/includes/site/TopBar';
import BrowseMenu from './components/includes/site/BrowseMenu';
import Confirmation from './components/utils/Confirmation';
import Alert from './components/utils/Alert';
import Prompt from './components/utils/Prompt';
import Warning from './components/utils/Warning';
import { ToggleMenu } from './actions/MenuActions';

class App extends Component {
	constructor(props) {
		super(props);

		this.state = {
			mainMenu: false
		}
	}

	componentDidUpdate(prevProps, prevState) {
		if (prevProps.location.key !== this.props.location.key) {
			this.props.dispatch(GetSession());
			this.props.dispatch(GetSectors());
		}
	}

	componentDidMount() {
		this.props.dispatch(GetSession());
		this.props.dispatch(GetSectors());

		document.body.addEventListener('click', (e) => {
			if (typeof e.target.className === 'object' || e.target.classList.contains('admin-menu-button') || e.target.classList.contains('menu-item') || e.target.classList.contains('menu') || e.target.id === 'browse-menu-button') {
				return;
			} else if (this.props.menu.open) {
				this.props.dispatch(ToggleMenu('', ''));
			}
		});
    }

	render() {
		let confirmation, sectors, alerts, prompt, warning;

		if (this.props.confirmation.status === true) {
			confirmation = <Confirmation message={this.props.confirmation.message} note={this.props.confirmation.note} />
		}

		if (this.props.sectors) {
			sectors = this.props.sectors.map((sector, i) => {
				return <Route key={i} path={`/sectors/${sector.sector.toLowerCase()}`} render={() => <Pages.Sectors name={sector.sector} />} />;
			});
		}

		if (this.props.menu.open === 'main') {
			this.menu = <BrowseMenu sectors={this.props.sectors} hide={false} />
		} else {
			this.menu = <BrowseMenu sectors={this.props.sectors} hide={true} />
		}

		if (this.props.alerts.length > 0) {
			alerts = this.props.alerts.map((alert, i) => {
				return <Alert key={i} status={alert.status} message={alert.message} unmount={() => {
					this.props.alerts.splice(0, 1);
					this.props.dispatch(RemoveAlert(this.props.alerts))
				}} />
			});
		} else if (this.props.alerts.length === 0) {
			alerts = [];
		}

		if (this.props.prompt.text) {
			prompt = <Prompt text={this.props.prompt.text} data={this.props.prompt.data} />;
		}

		if (this.props.warning.message) {
			warning = <Warning message={this.props.warning.message} />;
		}

		return (
			<div className='col-container'>
				{warning}
				{prompt}
				{confirmation}
				<TopBar />

				<div className='position-relative'>{this.menu}</div>

				<section className='main-container'>
					<Switch>
						<Route exact path='/' component={Pages.Main} />
						<Route exact path='/view' component={Pages.ViewUser} />
						<Route exact path='/dashboard/list' render={() => <Pages.Dashboard user={this.props.user}><Pages.ListSettings user={this.props.user} /></Pages.Dashboard>} />
						<Route exact path='/dashboard/saved_listings' render={() => <Pages.Dashboard user={this.props.user}><Pages.SavedListings user={this.props.user} /></Pages.Dashboard>} />
						<Route exact path='/dashboard/edit' render={() => <Pages.Dashboard user={this.props.user}><Pages.EditUser user={this.props.user} /></Pages.Dashboard>} />
						<Route exact path='/dashboard/settings' render={() => <Pages.Dashboard user={this.props.user}><Pages.UserSettings user={this.props.user} /></Pages.Dashboard>} />
						<Route exact path='/dashboard/messages/:stage' render={() => <Pages.Dashboard user={this.props.user}><Pages.Messages user={this.props.user} /></Pages.Dashboard>} />} />
						<Route exact path='/dashboard/message/:stage/:id/details' render={() => <Pages.Dashboard user={this.props.user}><Pages.MessageDetails user={this.props.user} /></Pages.Dashboard>} />
						<Route exact path='/listing/:id' render={() => <Pages.ListingDetails user={this.props.user} />} />
						<Route exact path='/user/:username' render={() => <Pages.ViewUser user={this.props.user} />} />
						<Route exact path='/account/login' render={() => <Pages.Login user={this.props.user} />} />
						<Route exact path='/account/register' render={() => <Pages.Register user={this.props.user} />} />
						{sectors}
						<Route exact path='/sectors/:sector' component={Pages.Sectors} />

						<Route exact path='/admin-panel' render={() => <Admin.Admin><Admin.AdminOverview user={this.props.user} /></Admin.Admin>} />
						<Route exact path='/admin-panel/sectors' render={() => <Admin.Admin><Admin.AdminSectors user={this.props.user} /></Admin.Admin>} />
						<Route exact path='/admin-panel/users' render={() => <Admin.Admin><Admin.AdminUsers user={this.props.user} /></Admin.Admin>} />
						<Route exact path='/admin-panel/listings' render={() => <Admin.Admin><Admin.AdminListings user={this.props.user} sectors={this.props.sectors} /></Admin.Admin>} />

						<Route render={() => <Pages.Response code={404} header={'Not Found'} message={`This page you're trying to access does not exist.`} />} />
					</Switch>
				</section>

				<div className='alert-container'>{alerts}</div>
			</div>
		);
	}
}

const mapStateToProps = (state) => {
	return {
		user: state.Login,
		confirmation: state.Confirmation,
		sectors: state.Sectors.sectors,
		alerts: state.Alert.alerts,
		prompt: state.Prompt,
		warning: state.Warning,
		menu: state.Menu
	}
}

export default withRouter(connect(mapStateToProps)(App));
