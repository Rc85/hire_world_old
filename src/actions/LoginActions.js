import fetch from 'axios';
import { Alert } from '../actions/AlertActions';
import { LogError } from '../components/utils/LogError';
import { GetSession } from './FetchActions';

export const LoginUser = (data) => {
    return dispatch => {
        document.body.style.overflow = 'hidden';
        dispatch(LoginBegin('login begin'));

        return fetch.post('/api/auth/login', data)
        .then(resp => {
            document.body.style.overflow = 'auto';
            if (resp.data.status === 'success') {
                dispatch(LoginSuccess(resp.data.status, resp.data.user));
            } else if (resp.data.status === 'error') {
                dispatch(LoginError(resp.data.status, resp.data.statusMessage));
                dispatch(Alert(resp.data.status, resp.data.statusMessage));
            } else if (resp.data.status === 'access error') {
                dispatch(LoginError(resp.data.status, resp.data.statusMessage));
            }
        })
        .catch(err => LogError(err, '/api/auth/login'));   
    }
}

const LoginBegin = (status) => {
    return {
        type: 'LOGIN_USER',
        status
    }
}

const LoginSuccess = (status, user) => {
    return {
        type: 'LOGIN_USER_UPDATE',
        user,
        status
    }
}

const LoginError = (status, statusMessage) => {
    return {
        type: 'LOGIN_USER_ERROR',
        status,
        statusMessage, 
        user: null
    }
}

export const LogoutUser = () => {
    return dispatch => {
        fetch.post('/api/auth/logout')
        .then(resp => {
            if (resp.data.status === 'error') {
                dispatch(Logout('not logged in'));
            }
        })
        .catch(err => LogError(err, '/api/auth/logout'));
    }
}

const Logout = (status) => {
    return {
        type: 'LOGOUT_USER',
        status
    }
}

export const UpdateUser = user => {
    return {
        type: 'USER_UPDATE',
        user
    }
}

export const ResetLogin = () => {
    return {
        type: 'LOGIN_USER_ERROR',
        status: null,
        statusMessage: null
    }
}