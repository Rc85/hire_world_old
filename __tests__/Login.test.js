import React from 'react';
import Enzyme, { mount, render, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import { MemoryRouter } from 'react-router';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import App from '../src/App';
import { LoginUser } from '../src/actions/LoginActions';
import { User } from '../__mocks__/userMock';
import moxios from 'moxios';
import { createStore, applyMiddleware } from 'redux';
import { reducers } from '../src/reducers';
import { GetSession } from '../src/actions/FetchActions';

const mockStore = configureStore([thunk]);
const store = createStore(reducers, applyMiddleware(thunk));

Enzyme.configure({adapter: new Adapter()});
jest.setTimeout(30000);

describe('Login page', () => {
    it('Matches snapshot', (done) => {
        const wrapper = shallow(
            <Provider store={store}><MemoryRouter initialEntries={[{pathname: '/', key: 'login'}]}><App /></MemoryRouter></Provider>
        );
        expect(toJson(wrapper)).toMatchSnapshot();

        done();
    });

    describe('clicking on body', () => {
        test('Browse menu should appear', (done) => {
            const wrapper = mount(
                <Provider store={store}><MemoryRouter initialEntries={[{pathname: '/', key: 'login'}]}><App /></MemoryRouter></Provider>
            );
            wrapper.find('#browse-menu-button').simulate('click');
            expect(wrapper.exists('#browse-menu')).toEqual(true);

            done();
        });
    });

    describe('Login', () => {
        beforeEach(() => {
            moxios.install();
        });

        afterEach(() => {
            moxios.uninstall();
        });

        test('Login in with existing credentials', async(done) => {
            moxios.stubRequest('/api/auth/login', {
                status: 200,
                response: {
                    status: 'get session success',
                    user: User
                }
            });
            
            const expectedActions = [
                {
                    type: 'LOGIN_USER',
                    status: 'getting session'
                }
            ];
            
            const store = mockStore({status: '', user: null});
            
            await store.dispatch(LoginUser({username: 'fakeuser', password: '111111'}))
            .then(() => {
                expect(store.getActions()[0]).toEqual(expectedActions[0]);
                done();
            });
        });
    });

    describe('Get session', () => {
        beforeEach(() => {
            moxios.install();
        });

        afterEach(() => {
            moxios.uninstall();
        });

        test('Get session should be successful', async(done) => {
            moxios.stubRequest('/api/auth/login', {
                status: 200,
                response: {
                    status: 'get session success',
                    user: User
                }
            });

            const expectedActions = [
                {
                    type: 'LOGIN_USER',
                    status: 'getting session'
                },
                {
                    type: 'LOGIN_USER_UPDATE',
                    status: 'get session success',
                    user: User
                }
            ];

            const store = mockStore({status: '', user: null});

            await store.dispatch(GetSession())
            .then(() => {
                expect(store.getActions()[0]).toEqual(expectedActions[0]);
                done();
            });
        });
    });
});