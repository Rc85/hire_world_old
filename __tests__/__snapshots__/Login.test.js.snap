// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login page Matches snapshot 1`] = `
<MemoryRouter
  initialEntries={
    Array [
      Object {
        "key": "login",
        "pathname": "/",
      },
    ]
  }
>
  <withRouter(Connect(App)) />
</MemoryRouter>
`;
